docs/dev/Pagination-v2.md

```diff
@@ -183,20 +573,75 @@ sequenceDiagram
     participant QueryPlanFactory
     participant CanPaginateVisitor
 
-RestSQLQueryAction->>+SQLService: prepareRequest
-  SQLService->>+QueryPlanFactory: execute
-    critical V2 support check
-    QueryPlanFactory->>+CanPaginateVisitor: canConvertToCursor
-      CanPaginateVisitor-->>-QueryPlanFactory: false
-    QueryPlanFactory-->>-RestSQLQueryAction: UnsupportedCursorRequestException
+RestSQLQueryAction ->>+ SQLService : prepareRequest
+  SQLService ->>+ QueryPlanFactory : execute
+    rect rgb(191, 223, 255)
+    note over SQLService, CanPaginateVisitor : V2 support check
+    QueryPlanFactory ->>+ CanPaginateVisitor : canConvertToCursor
+      CanPaginateVisitor -->>- QueryPlanFactory : false
+    QueryPlanFactory -->>- RestSQLQueryAction : UnsupportedCursorRequestException
     deactivate SQLService
     end
-      RestSQLQueryAction->>Legacy Engine: accept
-      Note over Legacy Engine: Processing in Legacy engine
-        Legacy Engine-->>RestSQLQueryAction:complete
+      RestSQLQueryAction ->> Legacy Engine: accept
+      Note over Legacy Engine : Processing in Legacy engine
+        Legacy Engine -->> RestSQLQueryAction : complete
 ```
 
-### Serialization
+#### Serialization and Deserialization round trip
+
+The SQL engine should be able to completely recover the Physical Plan tree to continue its execution to get the next page. Serialization mechanism is responsible of proper dumping the plan tree. `ResourceMonitorPlan` shouldn't be serialized, because a new object of this type would be created for the restored plan tree before execution.
```

Comment:
```suggestion
The SQL engine should be able to completely recover the Physical Plan tree to continue its execution to get the next page. Serialization mechanism is responsible for recovering the plan tree. note: `ResourceMonitorPlan` isn't serialized, because a new object of this type would be created for the restored plan tree before execution. 
```